docs/solidity-to-cadence.mdx

```diff
@@ -0,0 +1,447 @@
+[Capability Bootstrapping](design-patterns#capability-bootstrapping) pattern for making the Capability available to
+other accounts.
+
+An example admin role implementation is available in [Caadence cookbook](https://cookbook.onflow.org/?preview=13).
+
+### Role-based access
+
+Implementing role-based-access can be achieved by defining roles implemented as Resources managed by the
+root-admin account. Roles can provide limited access to functions which guard other protected resources, with access
+levels and/or what is exposed varying from role to role. The root-admin can grant accounts access to individual
+roles through a private capability. Functions that the roles are permitted to invoke may be scoped as
+`access(contract)` to enforce that they can only be called by code paths in the root-admin contract.
+
+# Other best practices and conventions
+
+Certain well established best practices for Solidity may not apply or are handled differently.
+
+## Check effects interactions
+
+Solidity contracts must use the [check effect interaction](https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html)
+because functions are public by default and address-based access means that guards must exist when program flow
+concedes control to an external contract. There are two reasons why this is significantly less of a problem in
+Cadence. Functions are private by default and the language provides a range of [access scopes](language#access-control).
+More importantly, 'risks associated with ceding control to an external contract' is an Ethereum phenomenon;
+the risk no longer applies because access control is handled with Capabilities while assets remain stored safely
+in accounts.
+
+## Guard Check
+
+Solidity uses `revert`, `require` & `assert` to validate inputs. `require` is a product of the address-based nature
+of Solidity which Capabilities replace. `revert` is similar to Cadence's `panic` in that a transaction is aborted.
+Cadence provides an `assert` operator which mirrors `assert` in Solidity.
+
+## Modifiers
+
+Modifiers are extensively used in Solidity when enforcing pre-checks within a function. This is a powerful language
+feature. However, modifiers can also mutate state which introduces risks to program control flow.
+
+Cadence uses `pre` and `post` blocks to validate input values or the function execution outputs. Notably, `pre` and
+`post` block prohibit changing of state and may only enforce conditions.
+
+Another difference is that modifiers in Solidity can be re-used within the contract multiple times. Cadence
+`pre` & `post` blocks are associated to individual functions only, reducing the likelihood of errors but which
+results in a small amount of code duplication.
+
+## Error handling
+
+Solidity offers try/catch block to handle errors, however, there is no equivalent in Cadence. This is because
+Resources and Capabilities codify the complexity of access and security in first-class types and
```

Comment:
Missing finish of the sentence. Anyway I'll maybe skip this point since I believe there's a chance that we get some kind of try / catch in cadence in the future (not really sure about this, just a vague memory I have about some conversations)

Reply:
Good catch, will clean it up. Thx